src/backend/distributed/operations/citus_tools.c

```diff
@@ -509,7 +509,11 @@ ExecuteRemoteQueryOrCommand(char *nodeName, uint32 nodePort, char *queryString,
 		return false;
 	}
 
-	SendRemoteCommand(connection, queryString);
+	if (!SendRemoteCommand(connection, queryString))
+	{
+		ReportConnectionError(connection, ERROR);
```

Comment:
I am not so sure if this is a safe to do. In case of connection errors, the user experience may change.

Reply:
I guess here we rely on returning a boolean to determine the success. Erroring out might break that.

To be on the safe side, we can do what is done above:

```
		appendStringInfo(queryResultString, "failed to send query to %s:%d", nodeName,
						 (int) nodePort);
		return false;
```